sdk/eventhub/Azure.Messaging.EventHubs/CONTRIBUTING.md

```diff
@@ -27,9 +25,118 @@ Tests in the Event Hubs client library are split into two categories:
+
+```powershell
+Connect-AzAccount -Subscription '<< AZURE SUBSCRIPTION ID >>'
+
+<repository-root>/eng/common/TestResources/New-TestResources.ps1 `
+    -BaseName '<< MEMORABLE VALUE (example: azsdk) >>' `
+    -ServiceDirectory 'eventhub' `
+    -SubscriptionId '<< AZURE SUBSCRIPTION ID >>' `
+    -ResourceGroupName '<< NAME FOR RESOURCE GROUP >>' `
+    -Location '<< AZURE REGION CODE (example: eastus) >>' `
+    -ArmTemplateParameters @{ perTestExecutionLimitMinutes = '5' }
+```
+
+The `ResourceGroupName` and `Location` parameters are optional; if not provided, the `BaseName` will be used to generate the name of the resource group and will be located in `westus2`.  For a list of the Azure locations valid for Event Hubs, the following command can be used:
+
+```powershell
+Get-AzLocation `
+    | Where { $_.Providers.Contains("Microsoft.EventHub") } `
+    | Select DisplayName, Location `
+    | Format-Table
+```
+
+If you prefer to use an existing service principal with the Live tests, the following set of parameters will instruct the script to create a new resource group for the tests and assign the "Owner" role for the resource group to the specified `TestAppliaticationId`.
+
+```powershell
+Connect-AzAccount -Subscription '<< AZURE SUBSCRIPTION ID >>'
+
+<repository-root>/eng/common/TestResources/New-TestResources.ps1 `
+    -BaseName '<< MEMORABLE VALUE (example: azsdk) >>' `
+    -ServiceDirectory 'eventhub' `
+    -SubscriptionId '<< AZURE SUBSCRIPTION ID >>' `
+    -ResourceGroupName '<< NAME FOR RESOURCE GROUP >>' `
+    -Location '<< AZURE REGION CODE (example: eastus) >>' `
+    -TestApplicationId '<< APPLICATION ID OF THE AAD APPLICATION >>' `
+    -TestApplicationSecret '<< APPLICATION SECRET OF THE AAD APPLICATION >>' `
+    -ArmTemplateParameters @{ perTestExecutionLimitMinutes = '5' }`
+```
+
+The full set of options for `New-TestResources.ps1` can be found in the [New-TestResources.ps1 documentation](https://github.com/Azure/azure-sdk-tools/blob/main/eng/common/TestResources/New-TestResources.ps1.md).
+
+#### Manually creating resources
+
+It is also possible to manually generate the test resources.  To do so:
 
-The Live tests read information from the following environment variables:
+1) Create a new resource group
+1) Create a service principal or select an existing service principal to use with the tests
+1) Ensure that your service principal is granted "Owner" or "Contributor" on the resource group
+1) Deploy the [Event Hubs test resources ARM template](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/eventhub/test-resources.json) to the resource group
+1) Capture the output from the ARM template deployment to configure the environment
+
+### Configuring the environment
```

Comment:
Should this be a sub-bullet of the automated setup section?

Reply:
Maybe?   I split it out because it's something that applies outside of automation.  For example, I'm configuring the environment manually because I decided to deploy the template via the portal instead of the scripts.

Does that change make sense or do you still think it's not a stand-alone section?

Reply:
The thing that seemed odd to me was that the manual instructions are just before this. It seems like the automated flow should be uninterrupted.

Reply:
My thought was to break it down into "Creating resources" and then "Configuring the environment", given that the configuring section is pretty long thanks to the details and the manual creation part is so short that you can easily see the next section after it.    I fear that it would be really easy to overlook that we discuss the manual creation if we push it until after the configuring section.

Thoughts?

Reply:
I don't feel strongly either way. My vote would be to leave the manual section for last since it should be the edge case.

Reply:
Fair enough... I'll play with it a bit and see.

Reply:
I gave the flow an overhaul to accentuate the automated approach and push the manual flow down.   Does that read better to you?

Reply:
Looks great!